compiler/rustc_mir_transform/src/errors.rs

```diff
@@ -141,10 +167,25 @@ impl<'a> DecorateLint<'a, ()> for UnsafeOpInUnsafeFn {
         let desc = diag
             .handler()
             .expect("lint should not yet be emitted")
-            .eagerly_translate_to_string(self.details.label(), [].into_iter());
+            .eagerly_translate_to_string(self.details.clone().label(), [].into_iter());
         diag.set_arg("details", desc);
-        diag.span_label(self.details.span, self.details.label());
-        diag.note(self.details.note());
+        diag.span_label(self.details.clone().span, self.details.clone().label());
+        if let Some(help) = self.help {
+            diag.help(help);
+        }
+        match self.details.violation {
```

Comment:
Same holds here, this can be collapsed into a single `if let Some(note_missing_features) = self.self.details.note_missing_features()`.